### PR TITLE
Update docs on custom controls, constructor API, all control types

### PR DESCRIPTION
**Pull request by hjhart**
References issue https://github.com/jonspalmer/view_component_storybook/issues/54

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,9 +3,10 @@
 The ViewComponent::Storybook gem provides Ruby api for writing stories describing [View Components](https://github.com/github/view_component) and allowing them to be previewed and tested in [Storybook](https://github.com/storybookjs/storybook/) via its [Server](https://github.com/storybookjs/storybook/tree/next/app/server) support.
 
 ## Features
-* A Ruby DSL for writing Stories describing View Components
-* A Rails controller backend for Storybook Server compatible with Storybook Controls Addon parameters
-* Coming Soon: Rake tasks to watch View Components and Stories and trigger Storybook hot reloading
+
+- A Ruby DSL for writing Stories describing View Components
+- A Rails controller backend for Storybook Server compatible with Storybook Controls Addon parameters
+- Coming Soon: Rake tasks to watch View Components and Stories and trigger Storybook hot reloading
```
Comment (hjhart) [inline README.md:9]: Some changes were made by a markdown linter. `markdownlint` in VSCode. Happy to back these out if undesirable.

Comment (jonspalmer) [reply README.md:9]: I like the linting changes

File: README.md [hunk 2]
```diff
@@ -97,9 +103,74 @@ class ButtonComponentStories < ViewComponent::Storybook::Stories
...
+      color(:background_color, preset_colors: ["#8CDED0", "#F7F6F7", "#83B8FE"])
+      object(:additional_attributes, { "aria-label": "Button label"})
+      select(:size, [:sm, :md, :base, :lg], :base)
+      multi_select(:variants, [:rounded, :striped, :primary, :secondary], [:primary, :rounded])
+      radio(:size, [:sm, :md, :base, :lg], :base)
+      inline_radio(:size, [:sm, :md, :base, :lg], :base)
+      check(:variants, [:rounded, :striped, :primary, :secondary], [:primary, :rounded])
+      inline_check(:variants, [:rounded, :striped, :primary, :secondary], [:primary, :rounded])
+      array(:variants, "rounded, striped, primary, secondary", ",")
+      date(:expiration_date, Date.today)
```
Comment (jonspalmer) [inline README.md:149]: love this set of examples

File: README.md [hunk 3]
```diff
@@ -97,9 +103,74 @@ class ButtonComponentStories < ViewComponent::Storybook::Stories
 end
 ```
 
+### Constructor API
```
Comment (jonspalmer) [inline README.md:106]: I would write this section a bit more directly. Introduce the concept first, active voice and avoiding 'you' etc. The examples in the ViewComponent docs like slots are good to pattern match from: https://viewcomponent.org/guide/slots.html

"Stories accept ViewComponent constructor arguments derived from Storybook controls, enabling dynamic rendering of the story.

For example:
"

**Review by jonspalmer — COMMENTED** [review]
This is GREAT!

I'm not the documentation writing expert but I've found the simple guide in ViewComponents useful: https://viewcomponent.org/CONTRIBUTING.html#style-guidelines

writing in the active voice is particularly powerful, hard to do in my experience, but more interesting to read.

**Comment by hjhart** [comment]
@jonspalmer Added active voice to the constructor API. I think we'll want to remove the `controls` section in favor of the `constructor` API but I think that can wait for a future PR.

File: README.md [hunk 4]
```diff
@@ -97,9 +103,76 @@ class ButtonComponentStories < ViewComponent::Storybook::Stories
...
+
+ViewComponent stories provide a constructor API to provide flexibility in your ViewComponent initializers.
+
+Give a header class that looks like the following:
```
Comment (jonspalmer) [inline README.md:110]:
```suggestion
Given a header class that looks like the following:
```

File: README.md [hunk 5]
```diff
@@ -97,9 +103,76 @@ class ButtonComponentStories < ViewComponent::Storybook::Stories
...
+
+```ruby
+class HeaderComponent < ViewComponent::Base
+  def initialize(tag, bold: true)
+    @tag = tag
+    @arguments = arguments
```
Comment (jonspalmer) [inline README.md:116]:
```suggestion
    @bold = bold
```

File: README.md [hunk 6]
```diff
@@ -97,9 +103,76 @@ class ButtonComponentStories < ViewComponent::Storybook::Stories
...
+end
+```
+
+To render a HeaderComponent, use the `constructor` method.
+
+```ruby
+class HeaderComponentStories < ViewComponent::Storybook::Stories
+  constructor(
+    text("h1"),
+    bold: boolean()
```
Comment (jonspalmer) [inline README.md:127]:
```suggestion
    bold: boolean(true)
```

**Review by jonspalmer — COMMENTED** [review]
Looks great. A couple of minor suggestions and we should merge this.

**Comment by jonspalmer** [comment]
Can we please squash merge PRs in the future to maintain a cleaner main branch

**Comment by hjhart** [comment]
> Can we please squash merge PRs in the future to maintain a cleaner main branch

Yes, sorry about that! I noticed that after merging in and winced a bit. 

You can in fact limit the method that contributors can merge in work through github settings, if that interests you at all.

https://docs.github.com/en/github/administering-a-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests

**Pull request merged**